code/game/objects/structures/girders.dm

```diff
@@ -4,48 +4,56 @@
 	anchored = 1
 	density = 1
 	layer = 2
-	var/state = 0
+	var/state = 0 //state 0 = normal girder, state 1 = no idea, state 2 = reinforced girder, state 3 = displaced/dislodged girder
 	var/girderpasschance = 20 // percentage chance that a projectile passes through the girder.
 
 /obj/structure/girder/attackby(obj/item/W as obj, mob/user as mob, params)
 	add_fingerprint(user)
-	if(istype(W, /obj/item/weapon/wrench) && state == 0)
-		if(anchored && !istype(src,/obj/structure/girder/displaced))
-			playsound(src.loc, 'sound/items/Ratchet.ogg', 100, 1)
-			user << "<span class='notice'>Now disassembling the girder...</span>"
-			if(do_after(user,40))
-				if(!src.loc)
-					return
-				user << "<span class='notice'>You dissasembled the girder!</span>"
-				new /obj/item/stack/sheet/metal(get_turf(src))
-				qdel(src)
-		else if(!anchored)
-			if (!istype(src.loc, /turf/simulated/floor))
-				usr << "<span class='danger'>A floor must be present to secure the girder!</span>"
-				return
-			playsound(src.loc, 'sound/items/Ratchet.ogg', 100, 1)
-			user << "<span class='notice'>Now securing the girder...</span>"
+	if(istype(W, /obj/item/weapon/weldingtool) && state == 3)

```

Comment:
replace these ```state == 3``` Checks with ```state == GIRDER_DISPLACED```

At the top of this file you'll want to add
```
#define GIRDER_NORMAL 0
#define GIRDER_REINF_STRUTS 1
#define GIRDER_REINF 2
#define GIRDER_DISPLACED 3
```
And then replace all the other checks with these too.